pkg/target/manager.go

```diff
@@ -53,6 +54,8 @@ type Manager interface {
 	// This implicitly unsets seed target configuration
 	// It will also configure appropriate project and seed values if not already set
 	TargetShoot(ctx context.Context, name string) error
+	// TargetControlPlane sets the control plane target flag
```

Comment:
I would replace all these `Target` und `UnsetTarget` methods with:
```go
    SetTarget(ctx context.Context, kind TargetKind, name string) error
    UnsetTarget(ctx context.Context, kind TargetKind) error
```
and would define the TargetKind enum in the target package:
```go
type TargetKind int

const (
	TargetKindInvalid TargetKind = iota - 1
	TargetKindAll
	TargetKindGarden
	TargetKindProject
	TargetKindSeed
	TargetKindShoot
	TargetKindControlPlane
)

var ValidTargetKinds = [...]string{"All", "Garden", "Project", "Seed", "Shoot", "ControlPlane"}

func (k TargetKind) String() string {
	if !k.IsValid() {
		return "Invalid"
	}

	return ValidTargetKinds[k]

}

func (k TargetKind) IsValid() bool {
	return k >= 0 && int(k) < len(ValidTargetKinds)
}
```

Reply:
In this Proposal the TargetPattern is renamed to TargetAll because UnsetPattern would not make sense in this case but UnsetAll would reset everything to an empty target

Reply:
Yes let's keep this in mind when we refactor the target command. I'm not sure if we still need targetkinds in this case. All targetkinds will then bre represented by a target subcommand and alltargets will be the target root command.